drift/core/beamtransfer.py

```diff
@@ -831,190 +831,260 @@ def _generate_svdfiles(self, regen=False, skip_svd_inv=False):
         ## m, performing the SVD, combining the beams and then write out the
         ## results.
 
-        # For each `m` collect all the `m` sections from each frequency file,
-        # and write them into a new `m` file. Use MPI if available.
-        for mi in mpiutil.mpirange(self.telescope.mmax + 1):
+        m_list = np.arange(self.telescope.mmax + 1)
+        if mpiutil.rank0:
```

Comment:
Why are you checking if file exists for that m only on rank0? You could do that in parallel, each process is independent of the other/

Reply:
I think it's better to keep the filesystem interactions on a single node where possible. For a few reasons: the filesystem tends to be less reliable than the network between the nodes; and the filesystem is slow and many parallel operations on the same directory can cause performance issues, as I think all the requests get sent to the same metadata fs node for completion.

Reply:
Part of my motivation was stability, and part of it was that after we determine which m's need to be computed, we want to split *that* list between tasks. The performance gain from forming the initial m list using multiple tasks would be very small.

Reply:
ok. makes sense.